components/eam/src/physics/cam/ndrop.F90

```diff
@@ -1339,270 +1312,73 @@ subroutine activate_modal(wbar, sigw, wdiab, wminf, wmaxf, tair, rhoair,  &
    alpha=gravit*(latvap/(cpair*rh2o*tair*tair)-1._r8/(rair*tair))
    gamma=(1+latvap/cpair*dqsdt)/(rhoair*qs)
    etafactor2max=1.e10_r8/(alpha*wmaxf)**1.5_r8 ! this should make eta big if na is very small.
-
-   do m=1,nmode
-      if(volume(m).gt.1.e-39_r8.and.na(m).gt.1.e-39_r8)then
+! BJG g renamed to gthermfac, defined here since same for all modes in implementation
+   gthermfac=1._r8/(rhoh2o/(diff0*rhoair*qs)                                    &
+           +latvap*rhoh2o/(conduct0*tair)*(latvap/(rh2o*tair)-1._r8))
+! BJG wdiab = 0 whenever subroutine is called, so removed from argument list
+!      wnuc=wbar+wdiab
+   wnuc = wbar
+!  BJG  below is always true if this point is reached.
+!      if(wnuc.gt.0._r8)then
```

Comment:
Same here. Thanks for noticing these things to get rid of these unnecessary codes.